test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/utils/v3/OpenShift3NativeProjectUtils.java

```diff
@@ -74,5 +79,35 @@ public static IProject createProject(String name, String displayName, String des
 		
 		return createdProject;
 	}
+	
+	private static class CreateProjectWaitCondition extends AbstractWaitCondition{
+
+		private IProject createdProject;
+		private Connection connection;
+		private IProjectRequest request;
+		
+		
+		
+		public CreateProjectWaitCondition(Connection connection, IProjectRequest request) {
+			super();
+			this.connection = connection;
+			this.request = request;
+		}
+
+		@Override
+		public boolean test() {
+			try{
+				createdProject = (IProject) connection.createResource(request);
```

Comment:
I extracted this into WaitCondition. Now I can do "new WaitUntil(createProjectWaitCondition);" on line 66. When it successfully creates the project on first try, everything is fine, but when it fails, the wait condition is evaluated again and again until it succeeds or hits time limit.